TestMultiChequeSimulation sometimes panics

TestMultiChequeSimulation sometimes panics with the error below. This is very rare (in my experience it only happens every 1000-1500 test runs and only on slower machines or VMs).

```
--- FAIL: TestMultiChequeSimulation (3.04s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc53dc2]

goroutine 39 [running]:
testing.tRunner.func1(0xc0000e3500)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xd84600, 0x16a17b0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/ethersphere/swarm/swap.TestMultiChequeSimulation(0xc0000e3500)
	/home/user/swarm/swap/simulations_test.go:460 +0x6e2
testing.tRunner(0xc0000e3500, 0xf3c398)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	github.com/ethersphere/swarm/swap	3.109s
FAIL
```

Line 460 here corresponds to:
```go
if err := creditorPeer.Send(context.Background(), &testMsgBigPrice{}); err != nil {
	t.Fatal(err)
}
```

--- a/swap/simulations_test.go
+++ b/swap/simulations_test.go
@@ -419,7 +419,7 @@ func TestMultiChequeSimulation(t *testing.T) {
 	// get the testService for the creditor
 	creditorSvc := sim.Service("swap", creditor).(*testService)
 
-	var debLen, credLen int
+	var debLen, credLen, debSwapLen, credSwapLen int
 	timeout := time.After(10 * time.Second)
 	for {
 		// let's always be nice and allow a time out to be catched
@@ -430,14 +430,16 @@ func TestMultiChequeSimulation(t *testing.T) {
 		}
 		// the node has all other peers in its peer list
 		debitorSvc.swap.peersLock.Lock()
-		debLen = len(debitorSvc.swap.peers)
+		debSwapLen = len(debitorSvc.swap.peers)
+		debLen = len(debitorSvc.peers)
 		debitorSvc.swap.peersLock.Unlock()
 
 		creditorSvc.swap.peersLock.Lock()
-		credLen = len(creditorSvc.swap.peers)
+		credSwapLen = len(creditorSvc.swap.peers)
+		credLen = len(creditorSvc.peers)
 		creditorSvc.swap.peersLock.Unlock()
 
-		if debLen == 1 && credLen == 1 {
+		if debLen == 1 && credLen == 1 && debSwapLen == 1 && credSwapLen == 1 {
 			break
 		}
 		// don't overheat the CPU...

